api/v1alpha1/hardwareclassification_types.go

```diff
@@ -48,6 +48,31 @@ type HardwareCharacteristics struct {
 	Nic *Nic `json:"nic,omitempty"`
 	// +optional
 	Ram *Ram `json:"ram,omitempty"`
+	// +optional
+	Firmware *Firmware `json:"firmware,omitempty"`
+	// +optional
+	SystemVendor *SystemVendor `json:"systemVendor,omitempty"`
+}
+
+// Firmware contains firmware details extracted from the hardware profile
+type Firmware struct {
+	Bios Bios `json:"bios,omitempty"`
```

Comment:
```suggestion
	BIOS BIOS `json:"bios,omitempty"`
```

Reply:
We have removed the BIOS from the firmware and made it single level only.